### ISSUE TITLE
Handling trailing slashes that get omitted in event property

### ISSUE THREAD

**Issue by bahoo** [adamchainz/apig-wsgi#376]
### Python Version

3.9

### Package Version

Latest

### Description

Flagging this as somewhere between issue discussion and possible / proposed fix — I noticed in my project that, when requesting a particular URL with a trailing slash, `event['requestContext']['http']['path']` coming from AWS will omit the trailing slash. I lost a couple hours digging into redirect loop ( in Django, repeatedly trying to redirect to [the admin login screen with its trailing slash in the URL](https://github.com/django/django/blob/main/django/contrib/admin/sites.py#L269), but never getting there because AWS's event doesn't pass that final slash along ) before finally discovering the issue.

Interestingly, another event property, `event['rawPath']` will correctly pass along that trailing slash. Don't know if it is documented anywhere in AWS world why one ( the "raw" path ) would include it, and the other would omit it, but after confirming this discrepancy side by side, I switched to [another library](https://github.com/asottile/api-gateway-v2-to-wsgi) which I found does [use `rawPath`](https://github.com/asottile/api-gateway-v2-to-wsgi/blob/main/api_gateway_v2_to_wsgi.py#L84) and it worked without issue. 💁🏻‍♂️

This feels like AWS's bug to fix ultimately ( either in code for consistency's sake, or at least documentation; ideally both ), and no interest or need in pitting two open source projects against eachother, but using `rawPath` or some derivative could be a solution here.

Worth noting for my particular instance, Django has [a whole `APPEND_SLASH` setting](https://docs.djangoproject.com/en/4.1/ref/settings/#std-setting-APPEND_SLASH) and [middleware](https://docs.djangoproject.com/en/4.1/ref/middleware/#django.middleware.common.CommonMiddleware) to handle this, but the admin redirect logic I mentioned kicks in before this fallback does. Maybe my scenario is too narrow / niche to warrant a code change 💁🏻‍♂️ but the admin's a pretty prominent cornerstone of using Django to begin with, and, anyway anyway ... mostly wanted to get this issue out of just my own brain, and hope this might help save someone a couple hours in the future. 👋🏻

**Comment by bahoo** [comment]
As a trivial demonstration, I created a Lambda function ( Python 3.9 ) that just returns the event exactly as given to it:

```
import json

def lambda_handler(event, context):
    return event
```

Deploying & visiting a URL with a trailing slash, e.g. `/foo/` shows it off:

<img width="589" alt="trailing-slash" src="https://user-images.githubusercontent.com/430415/189529925-8cf02e2f-fad1-482b-9ea7-8e89f1b9864e.png">

**Comment by adamchainz** [comment]
Thanks for the report. This has been fixed in #426 and released in 2.16.0.
